llvm/lib/Target/AArch64/AArch64Subtarget.cpp

```diff
@@ -673,3 +673,83 @@ bool AArch64Subtarget::isX16X17Safer() const {
 bool AArch64Subtarget::enableMachinePipeliner() const {
   return getSchedModel().hasInstrSchedModel();
 }
+
+bool AArch64Subtarget::isRegInClass(const MachineInstr &MI, const Register &Reg,
+                                    const TargetRegisterClass *TRC) const {
+  if (Reg.isPhysical()) {
+    return TRC->contains(Reg);
+  }
+  const MachineRegisterInfo &MRI = MI.getMF()->getRegInfo();
+  return TRC->hasSubClassEq(MRI.getRegClass(Reg));
+}
+
+/// NOTE: must maintain consistency with `AArch64InstrInfo::copyPhysReg`.
+bool AArch64Subtarget::canLowerToZeroCycleRegMove(
```

Comment:
Functional changes should come with a test. If that's in one of the follow-up commits, you should probably squash it into this one.